pkg/controller/operands/operand_test.go

```diff
@@ -81,4 +87,69 @@ var _ = Describe("Test operator.go", func() {
 			Expect(obj.Spec.Config.FilesystemOverhead.Global).Should(BeEquivalentTo("55"))
 		})
 	})
+
+	Context("Test addCrToTheRelatedObjectList", func() {
+		It("Should return error when apiVersion, kind and name missing", func() {
+			hco := commonTestUtils.NewHco()
+			req := commonTestUtils.NewReq(hco)
+			found := &cdiv1beta1.CDI{}
+
+			operand := genericOperand{Scheme: scheme.Scheme}
+			err := operand.addCrToTheRelatedObjectList(req, found)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("object reference must have, at a minimum: apiVersion, kind, and name"))
+		})
+
+		It("Should add into the list when it is missing", func() {
+			hco := commonTestUtils.NewHco()
+			req := commonTestUtils.NewReq(hco)
+			found := &cdiv1beta1.CDI{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "CDI",
+					APIVersion: "cdi.kubevirt.io/v1beta1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cdi-kubevirt-hyperconverged",
+				},
+			}
+
+			operand := genericOperand{Scheme: scheme.Scheme}
+			err := operand.addCrToTheRelatedObjectList(req, found)
+			Expect(err).ToNot(HaveOccurred())
+
+			foundRef, err := reference.GetReference(operand.Scheme, found)
+			Expect(err).To(BeNil())
+			Expect(hco.Status.RelatedObjects).To(ContainElement(*foundRef))
+		})
+
+		It("Should update the list and set StatusDirty=true when the resourceVersion is different", func() {
+			hco := commonTestUtils.NewHco()
+			req := commonTestUtils.NewReq(hco)
+			found := &cdiv1beta1.CDI{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "CDI",
+					APIVersion: "cdi.kubevirt.io/v1beta1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "cdi-kubevirt-hyperconverged",
+					ResourceVersion: "111",
+				},
+			}
+
+			operand := genericOperand{Scheme: scheme.Scheme}
+			err := operand.addCrToTheRelatedObjectList(req, found)
+			Expect(err).ToNot(HaveOccurred())
+
+			// update resource version
+			found.ResourceVersion = "112"
+			err = operand.addCrToTheRelatedObjectList(req, found)
+			Expect(err).ToNot(HaveOccurred())
+
+			foundRef, err := reference.GetReference(operand.Scheme, found)
+			Expect(err).To(BeNil())
+			Expect(hco.Status.RelatedObjects).To(ContainElement(*foundRef))
+			Expect(req.StatusDirty).To(BeTrue())
```

Comment:
Please add a check that the "old" reference was replaced, and not exists after the update.

Reply:
```suggestion
			operand := genericOperand{Scheme: scheme.Scheme}
			err := operand.addCrToTheRelatedObjectList(req, found)
			Expect(err).ToNot(HaveOccurred())
                        oldRef, err := reference.GetReference(operand.Scheme, found)
			Expect(err).ToNot(HaveOccurred())
                        
			// update resource version
			found.ResourceVersion = "112"
			err = operand.addCrToTheRelatedObjectList(req, found)
			Expect(err).ToNot(HaveOccurred())

			newRef, err := reference.GetReference(operand.Scheme, found)
			Expect(err).ToNot(HaveOccurred())
			Expect(hco.Status.RelatedObjects).To(ContainElement(*newRef))
			Expect(hco.Status.RelatedObjects).ToNot(ContainElement(*oldRef))
			Expect(req.StatusDirty).To(BeTrue())
```